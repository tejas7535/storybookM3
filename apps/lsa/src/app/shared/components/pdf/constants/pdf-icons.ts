export const starsIconPrimary =
  'iVBORw0KGgoAAAANSUhEUgAAAGAAAABgCAYAAADimHc4AAAAAXNSR0IArs4c6QAABqpJREFUeF7tXVty3DYQJJVrxbF9qbjyFevL5VzK8uNa1qaAXaxAimT3vACoivxxWcJiBt0zPQOSws7TeXVFYO5q/TQ+nQR0DoKTgJOAzgh0Nn9mwElAZwQ6mz8z4CQAIPDlzw95xMPD9d/58v7+ict0+9n0lH92mb8vZnt+fpr++XX93aDXeBlQAP9j/ncqAFvBm6enTM6nH5+tU3l/fgwCEugJ8BzFt6j2XmmZr5AxSHb0JaAAHw36IZnzY8/M6EPAEMBXrHSUqLYEjAb8q8yYH6fG0tSGgOGBXzPRTpbiCUjgP8zfompq7LzxRMQS8PWvz9N0uXY32ivpc+nxkzzUV93jr/cL1w/ZbOc5YkmII8ACfgL99+Uxr9+6kcoZ+PAhb+DU3VYcCTEE/Pfum3ixBXQr4CjTtIGR/Pv750c0vfT3vgSoi21chO0CoiXi+fLRnJWLDlhK2dH4r+8usukcgE+kW7JGSoRzJvhlgER2vOQmgefVt0v9d5IjHwJEUeQQ9SXNEmjpcgJj6rAOOwEdnL7LXJG8Tz/t6yiTNl6PzXGJs87FaxGtkXOjoma0bSSALLpGJzcxqMl3LozZnmQHb1ifngA6+h01v2Zi3XF5ylCxw5JgCAAdAaxjUdv4LfINUXioMmygKe3rCKCcCor8hNaWfUMUIpmnuiOlfTkBDPhKZyAQ905lp/ZEyFDd8qJ7SYoskBPAbFgigTiSPwUANOmM7CoCT0YAE/1Rus/06QoAaAL2pG89gTAIhAQQbWdk9GcQDnyIJgDZT78X+sATwKRgdPQzAAgjUJQBAVnAE8DIT3j0E0/YoglIJKA6KMgCngBktEX0Qx/kEiDOAGaXHEIAutcfHf2M/BQ0R/CFzEQuA5D8CBhXRRyrvWVycvFqXxgZIn3gCICpH7jrZdrPNZItAgI1JaQPHAFIfki2TRGHfFhPHi1DiIDkD+EDJkBrqLynY0L99uH8twHSd3xurxl62E9zbD13RspABCYmAOn/XvejfkPCCzGneY6eXw9NgEa7nTDzmwbUNm1wVg7GZUCNAnTUDzK3mQj5wE/NcHNiJ4Bx9C1lA9m95CWh+kjMhQlw0LlFVA6dDThiF2sZggCi1XolC6MVaO2LYk0IQP23hoChJEkY9etoMuKDJchoABbFnpIkqV97CzHigwnwrgG7CyFuNUM2yQFEcSRnMj8gGoeAtOIm2WCUnJqZJjWgVQaUhUUVaG2hPUoFRADxjARnAIpKDx3dWiTSVlojbgMj/ETYNCGAMCLFKkSK3iwBKM08C1pke9rFT1xvsAR1IYB4/UWcVtz9edG0SCaJrMME5O4EAEIYohcGdZWe6fVATz9RYCbrxCaVI6BlJxRJgKcMIT9JWxwBiG3SGBW7KNuoSXYGefqJgpJsTnwIINMNYoeIhhMQA7xkCAUKIT/JW46ANBIx7rEwlNY1vvVxBpLDQDyyAPkpsOFHgMDobpwikssHt2whUI4+SyTOYgiKflJ+ZBlAyQPuew/XCheWXd4/YowlwZKtjA3B/HwGMDJkqQVeC/OaZy9SmCAh9V+WAWl0ZBYcAaeRN+/58n6IuWUuUwFZBrBZIEjBe6DtRpZsQUutPgBMEKXYx9sIRaDICWCyQOrIXmRpiFxLh9fcVIMgDxY5AWwWCDqBV6kdce9+DaAkSAKk56Upk7ZgdC0AHUttdyE/8iiil7AGkpEhCvxpmpTZqssAOgsIx2pJUy6CJmBdSJE9FnxJtq+c1RPA1IJi7GihaZHpQL10SJ/l5CsJC+WxZ/rM3llDkvUxmbTjn54AkRQdyFEioNep5nvHnUnAN0S/fB+wxSKdpoKaIIlm77GN12PLgLL4xk57Y/7S5zMbrXv/4nLqug8BkqJ8zbuQMzjVxEhfhXH0348AKQkZrcCWk2VDlL23SQ1Fd+2WLwGi4lW70oEIDfABG0RfAgqm1LZ9K0QbECGVm7vkx8hmDAHrDQ8rB/Vi05fueBzKWv5a0/SlQHGBEUeAlYQFafP1JPV0lSPst46uL2PKaenp/+iUKxgcceBf+5HoS6O10T6x86NbFew8B+PiCSjG3xIRAcV2j4N2BCQPcpek+at3h1BjpmgI/Eu5YxzzHlNuwJn12cmxDsD3JWAUWeoI/BgEFC+KNJm+54XMhgR6ulre/h6iCJP45GHeElW/RdfqmQO53rZFmHRqMexeuKuflq+0XX+dbRpSvtJ2a7+gsR/8mfEJCAag9/QnAZ0ZOAk4CeiMQGfzZwacBHRGoLP5MwNOAjoj0Nn8/w48Z50ZjVGnAAAAAElFTkSuQmCC';
