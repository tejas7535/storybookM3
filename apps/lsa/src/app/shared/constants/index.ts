export * from './language';
export * from './lubricant-type.enum';
export * from './lubrication-points.enum';
export * from './medium.enum';
export * from './optime.enum';
export * from './pages.enum';
export * from './power-supply.enum';
export * from './relubrication-interval.enum';
