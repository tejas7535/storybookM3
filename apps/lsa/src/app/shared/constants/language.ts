import { LangDefinition } from '@jsverse/transloco/lib/types';

export const LANGUAGE_STORAGE_KEY = 'language';

export const LANGUAGE_EN: LangDefinition = {
  id: 'en',
  label: 'English',
};

export const LANGUAGE_DE: LangDefinition = {
  id: 'de',
  label: 'Deutsch',
};

export const LANGUAGE_FR: LangDefinition = {
  id: 'fr',
  label: 'Français',
};

export const LANGUAGE_ES: LangDefinition = {
  id: 'es',
  label: 'Español',
};

export const LANGUAGE_IT: LangDefinition = {
  id: 'it',
  label: 'Italiano',
};

export const LANGUAGE_ZH: LangDefinition = {
  id: 'zh',
  label: '简体中文',
};

export const LANGUAGE_ZH_TW: LangDefinition = {
  id: 'zh_TW',
  label: '繁體中文',
};

export const LANGUAGE_TR: LangDefinition = {
  id: 'tr',
  label: 'Türkçe',
};

export const LANGUAGE_JA: LangDefinition = {
  id: 'ja',
  label: '日本語',
};

export const LANGUAGE_RU: LangDefinition = {
  id: 'ru',
  label: 'Русский',
};

export const LANGUAGE_PL: LangDefinition = {
  id: 'pl',
  label: 'Polski',
};

export const LANGUAGE_PT: LangDefinition = {
  id: 'pt',
  label: 'Português',
};

export const LANGUAGE_KO: LangDefinition = {
  id: 'ko',
  label: '한국인',
};

export const LANGUAGE_VI: LangDefinition = {
  id: 'vi',
  label: 'Tiếng Việt',
};

export const LANGUAGE_CS: LangDefinition = {
  id: 'cs',
  label: 'Čeština',
};

export const LANGUAGE_TH: LangDefinition = {
  id: 'th',
  label: 'ไทย',
};

export const AVAILABLE_LANGUAGES: LangDefinition[] = [
  LANGUAGE_EN,
  LANGUAGE_DE,
  LANGUAGE_ES,
  LANGUAGE_FR,
  LANGUAGE_IT,
  LANGUAGE_ZH,
  LANGUAGE_ZH_TW,
  LANGUAGE_TR,
  LANGUAGE_JA,
  LANGUAGE_RU,
  LANGUAGE_PL,
  LANGUAGE_PT,
  LANGUAGE_KO,
  LANGUAGE_VI,
  LANGUAGE_CS,
  LANGUAGE_TH,
];

export const FALLBACK_LANGUAGE: LangDefinition = LANGUAGE_EN;
