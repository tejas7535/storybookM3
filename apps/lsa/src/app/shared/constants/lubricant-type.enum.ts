export enum LubricantType {
  Arcanol = 'Arcanol',
  Grease = 'Grease',
  Oil = 'Oil',
}
