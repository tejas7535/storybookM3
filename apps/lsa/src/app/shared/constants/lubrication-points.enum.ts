export enum LubricationPoints {
  One = '1',
  TwoToFour = '2-4',
  FiveToEight = '5-8',
  EightOrMore = '>8',
}
