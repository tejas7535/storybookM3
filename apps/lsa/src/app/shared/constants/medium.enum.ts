export enum Medium {
  Grease = 'Grease',
  Oil = 'Oil',
}
