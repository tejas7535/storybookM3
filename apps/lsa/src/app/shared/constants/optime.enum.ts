export enum Optime {
  No = 0,
  Yes = 1,
  NoPreference = 2,
}
