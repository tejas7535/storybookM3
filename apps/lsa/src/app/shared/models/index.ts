export * from './add-to-cart.model';
export * from './grease.model';
export * from './page.model';
export * from './recommendation-form.model';
export * from './recommendation-form-value.model';
export * from './recommendation-request.model';
export * from './recommendation-response.model';
export * from './recommendation-table-data.model';
