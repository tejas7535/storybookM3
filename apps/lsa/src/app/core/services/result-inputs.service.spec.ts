import { combineLatest, of } from 'rxjs';

import { TranslocoService } from '@jsverse/transloco';
import { LubricantType } from '@lsa/shared/constants';
import { RecommendationFormValue } from '@lsa/shared/models';
import { ResultInputModel } from '@lsa/shared/models/result-inputs.model';
import {
  mockApplicationInput,
  mockLubricantInput,
  mockLubricationPointsInput,
} from '@lsa/testing/mocks/input.mock';
import { createServiceFactory, SpectatorService } from '@ngneat/spectator/jest';

import { provideTranslocoTestingModule } from '@schaeffler/transloco/testing';

import { ResultInputsService } from './result-inputs.service';

describe('ResultInputsService', () => {
  let spectator: SpectatorService<ResultInputsService>;

  const createService = createServiceFactory({
    service: ResultInputsService,
    imports: [provideTranslocoTestingModule({ en: {} })],
    mocks: [TranslocoService],
  });

  beforeEach(() => {
    spectator = createService();
    const translocoService = spectator.inject(TranslocoService);
    translocoService.selectTranslate = jest.fn((key: string) =>
      of(`translated: ${key}`)
    ) as any;
  });

  it('should be created', () => {
    expect(spectator.service).toBeTruthy();
  });

  describe('when getting result inputs', () => {
    let mockFormValue: RecommendationFormValue;
    let result: ResultInputModel;

    beforeEach(() => {
      // eslint-disable-next-line unicorn/numeric-separators-style
      mockFormValue = {
        lubricationPoints: mockLubricationPointsInput,
        lubricant: mockLubricantInput,
        application: mockApplicationInput,
      };
      result = spectator.service.getResultInputs(mockFormValue);
    });

    it('should return title observable translation', (done) => {
      combineLatest([
        result.sections[0].title$,
        result.sections[1].title$,
        result.sections[2].title$,
      ]).subscribe(
        ([lubricationPointsTitle, lubricantTitle, applicationTitle]) => {
          expect(lubricationPointsTitle).toMatchSnapshot();
          expect(lubricantTitle).toMatchSnapshot();
          expect(applicationTitle).toMatchSnapshot();
          done();
        }
      );
    });

    // getting timeout on test, need to investigate after release
    it.skip('should return lubrication Points observable translation', (done) => {
      result.sections[0].inputs$.subscribe((lubricationPointsInputs) => {
        expect(lubricationPointsInputs).toMatchSnapshot();
        done();
      });
    });

    it('should return Lubricant observable translation', (done) => {
      result.sections[1].inputs$.subscribe((lubricationPointsInputs) => {
        expect(lubricationPointsInputs).toMatchSnapshot();
        done();
      });
    });

    it('should return application observable translation', (done) => {
      result.sections[2].inputs$.subscribe((lubricationPointsInputs) => {
        expect(lubricationPointsInputs).toMatchSnapshot();
        done();
      });
    });
  });

  describe('when getting non arcanol inputs', () => {
    let mockFormValue: RecommendationFormValue;
    let result: ResultInputModel;

    beforeEach(() => {
      mockFormValue = {
        lubricationPoints: mockLubricationPointsInput,
        lubricant: {
          ...mockLubricantInput,
          lubricantType: LubricantType.Grease,
        },
        application: mockApplicationInput,
      };
      result = spectator.service.getResultInputs(mockFormValue);
    });

    it('should return title observable translation', (done) => {
      combineLatest([
        result.sections[0].title$,
        result.sections[1].title$,
        result.sections[2].title$,
      ]).subscribe(
        ([lubricationPointsTitle, lubricantTitle, applicationTitle]) => {
          expect(lubricationPointsTitle).toMatchSnapshot();
          expect(lubricantTitle).toMatchSnapshot();
          expect(applicationTitle).toMatchSnapshot();
          done();
        }
      );
    });

    // getting timeout on test, need to investigate after release
    it.skip('should return lubrication Points inputs observable translation', (done) => {
      result.sections[0].inputs$.subscribe((lubricationPointsInputs) => {
        expect(lubricationPointsInputs).toMatchSnapshot();
        done();
      });
    });

    it('should return Lubricant inputs observable translation', (done) => {
      result.sections[1].inputs$.subscribe((lubricationPointsInputs) => {
        expect(lubricationPointsInputs).toMatchSnapshot();
        done();
      });
    });

    it('should return application inputs observable translation', (done) => {
      result.sections[2].inputs$.subscribe((lubricationPointsInputs) => {
        expect(lubricationPointsInputs).toMatchSnapshot();
        done();
      });
    });
  });
});
