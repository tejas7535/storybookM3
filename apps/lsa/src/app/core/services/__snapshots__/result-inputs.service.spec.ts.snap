// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultInputsService when getting non arcanol inputs should return Lubricant inputs observable translation 1`] = `
Array [
  Object {
    "title": "translated: pages.lubricant.title",
    "value": "translated: recommendation.lubrication.options.grease",
  },
]
`;

exports[`ResultInputsService when getting non arcanol inputs should return application inputs observable translation 1`] = `
Array [
  Object {
    "title": "translated: inputs.temperature.title",
    "value": "translated: inputs.temperature.value",
  },
  Object {
    "title": "translated: inputs.powerSupplyTitle",
    "value": "translated: recommendation.application.powerOptions.external",
  },
]
`;

exports[`ResultInputsService when getting non arcanol inputs should return lubrication Points inputs observable translation 1`] = `
Array [
  Object {
    "title": "translated: inputs.lubricationPoints",
    "value": "2-4",
  },
  Object {
    "title": "translated: inputs.relubricationQuantity.title",
    "value": "translated: inputs.relubricationQuantity.value",
  },
  Object {
    "title": "translated: inputs.maxPipeLengthTitle",
    "value": "translated: recommendation.lubricationPoints.pipeLengthOptions.lessThan",
  },
  Object {
    "remoteValue": "translated: recommendation.lubricationPoints.optime.0",
    "title": "translated: inputs.optimeTitle",
    "value": "translated: recommendation.lubricationPoints.optime.0",
  },
]
`;

exports[`ResultInputsService when getting non arcanol inputs should return title observable translation 1`] = `"translated: pages.lubricationPoints.title"`;

exports[`ResultInputsService when getting non arcanol inputs should return title observable translation 2`] = `"translated: pages.lubricant.title"`;

exports[`ResultInputsService when getting non arcanol inputs should return title observable translation 3`] = `"translated: pages.application.title"`;

exports[`ResultInputsService when getting result inputs should return Lubricant observable translation 1`] = `
Array [
  Object {
    "title": "translated: inputs.lubricationPoints",
    "value": "2-4",
  },
  Object {
    "title": "translated: inputs.relubricationQuantity.title",
    "value": "translated: inputs.relubricationQuantity.value",
  },
  Object {
    "title": "translated: inputs.maxPipeLengthTitle",
    "value": "translated: recommendation.lubricationPoints.pipeLengthOptions.directMontage",
  },
  Object {
    "remoteValue": "translated: recommendation.lubricationPoints.optime.0",
    "title": "translated: inputs.optimeTitle",
    "value": "translated: recommendation.lubricationPoints.optime.0",
  },
]
`;

exports[`ResultInputsService when getting result inputs should return Lubricant observable translation 2`] = `
Array [
  Object {
    "title": "translated: pages.lubricant.title",
    "value": "Arcanol MULTI2",
  },
]
`;

exports[`ResultInputsService when getting result inputs should return application observable translation 1`] = `
Array [
  Object {
    "title": "translated: inputs.lubricationPoints",
    "value": "2-4",
  },
  Object {
    "title": "translated: inputs.relubricationQuantity.title",
    "value": "translated: inputs.relubricationQuantity.value",
  },
  Object {
    "title": "translated: inputs.maxPipeLengthTitle",
    "value": "translated: recommendation.lubricationPoints.pipeLengthOptions.directMontage",
  },
  Object {
    "remoteValue": "translated: recommendation.lubricationPoints.optime.0",
    "title": "translated: inputs.optimeTitle",
    "value": "translated: recommendation.lubricationPoints.optime.0",
  },
]
`;

exports[`ResultInputsService when getting result inputs should return application observable translation 2`] = `
Array [
  Object {
    "title": "translated: inputs.temperature.title",
    "value": "translated: inputs.temperature.value",
  },
  Object {
    "title": "translated: inputs.powerSupplyTitle",
    "value": "translated: recommendation.application.powerOptions.external",
  },
]
`;

exports[`ResultInputsService when getting result inputs should return lubrication Points observable translation 1`] = `
Array [
  Object {
    "title": "translated: inputs.lubricationPoints",
    "value": "2-4",
  },
  Object {
    "title": "translated: inputs.relubricationQuantity.title",
    "value": "translated: inputs.relubricationQuantity.value",
  },
  Object {
    "title": "translated: inputs.maxPipeLengthTitle",
    "value": "translated: recommendation.lubricationPoints.pipeLengthOptions.directMontage",
  },
  Object {
    "remoteValue": "translated: recommendation.lubricationPoints.optime.0",
    "title": "translated: inputs.optimeTitle",
    "value": "translated: recommendation.lubricationPoints.optime.0",
  },
]
`;

exports[`ResultInputsService when getting result inputs should return title observable translation 1`] = `"translated: pages.lubricationPoints.title"`;

exports[`ResultInputsService when getting result inputs should return title observable translation 2`] = `"translated: pages.lubricant.title"`;

exports[`ResultInputsService when getting result inputs should return title observable translation 3`] = `"translated: pages.application.title"`;

exports[`ResultInputsService when getting result inputs when remote optime returns error should return lubrication Points observable translation based on form value 1`] = `
Array [
  Object {
    "title": "translated: inputs.lubricationPoints",
    "value": "2-4",
  },
  Object {
    "title": "translated: inputs.relubricationQuantity.title",
    "value": "translated: inputs.relubricationQuantity.value",
  },
  Object {
    "title": "translated: inputs.maxPipeLengthTitle",
    "value": "translated: recommendation.lubricationPoints.pipeLengthOptions.directMontage",
  },
  Object {
    "remoteValue": "translated: recommendation.lubricationPoints.optime.0",
    "title": "translated: inputs.optimeTitle",
    "value": "translated: recommendation.lubricationPoints.optime.0",
  },
]
`;

exports[`ResultInputsService when getting result inputs when remote optime returns error should return lubrication Points observable translation based on form value 2`] = `
Array [
  Object {
    "title": "translated: inputs.lubricationPoints",
    "value": "2-4",
  },
  Object {
    "title": "translated: inputs.relubricationQuantity.title",
    "value": "translated: inputs.relubricationQuantity.value",
  },
  Object {
    "title": "translated: inputs.maxPipeLengthTitle",
    "value": "translated: recommendation.lubricationPoints.pipeLengthOptions.directMontage",
  },
  Object {
    "remoteValue": "translated: recommendation.lubricationPoints.optime.0",
    "title": "translated: inputs.optimeTitle",
    "value": "translated: recommendation.lubricationPoints.optime.0",
  },
]
`;

exports[`ResultInputsService when getting result inputs when remote optime returns error should return lubrication Points observable translation based on form value 3`] = `
Array [
  Object {
    "title": "translated: inputs.lubricationPoints",
    "value": "2-4",
  },
  Object {
    "title": "translated: inputs.relubricationQuantity.title",
    "value": "translated: inputs.relubricationQuantity.value",
  },
  Object {
    "title": "translated: inputs.maxPipeLengthTitle",
    "value": "translated: recommendation.lubricationPoints.pipeLengthOptions.between",
  },
  Object {
    "remoteValue": "translated: recommendation.lubricationPoints.optime.0",
    "title": "translated: inputs.optimeTitle",
    "value": "translated: recommendation.lubricationPoints.optime.0",
  },
]
`;
